trollsift/parser.py

```diff
@@ -339,35 +376,46 @@ def _get_number_from_fmt(fmt):
 
 def _convert(convdef, stri):
     """Convert the string *stri* to the given conversion definition *convdef*."""
+    is_fixed_point = any([ftype in convdef for ftype in fixed_point_types])
     if '%' in convdef:
         result = dt.datetime.strptime(stri, convdef)
-    elif 'd' in convdef or 's' in convdef:
-        regex_match = fmt_spec_regex.match(convdef)
-        match_dict = regex_match.groupdict() if regex_match else {}
-        align = match_dict.get('align')
-        pad = match_dict.get('fill')
-        if align:
-            # align character is the last one
-            align = align[-1]
-        if align and align in '<>^' and not pad:
-            pad = ' '
-
-        if align == '>':
-            stri = stri.lstrip(pad)
-        elif align == '<':
-            stri = stri.rstrip(pad)
-        elif align == '^':
-            stri = stri.strip(pad)
-
+    elif 'd' in convdef or 's' in convdef or is_fixed_point:
+        stri = _strip_padding(convdef, stri)
         if 'd' in convdef:
             result = int(stri)
+        elif is_fixed_point:
+            result = float(stri)
         else:
             result = stri
     else:
         result = stri
     return result
 
 
+def _strip_padding(convdef, stri):
+    """Strip padding from the given string.
+
+    Args:
+        stri: String to be modified
+        convdef: Corresponding conversion definition (indicates the padding)
```

Comment:
Args are flipped in the docstring compared to the definition.

Reply:
Resolved, thanks!